python/python/lance/dataset.py

```diff
@@ -1440,6 +1454,24 @@ def create_index(
                 )
             kwargs["num_partitions"] = num_partitions
 
+            if (precomputed_partiton_dataset is not None) and (ivf_centroids is None):
+                raise ValueError(
+                    "ivf_centroids must be provided when"
+                    " precomputed_partiton_dataset is provided"
+                )
+            if precomputed_partiton_dataset is not None:
+                precomputed_ds = LanceDataset(precomputed_partiton_dataset)
+                if len(precomputed_ds.get_fragments()) != 1:
+                    raise ValueError(
+                        "precomputed_partiton_dataset must have only one fragment"
+                    )
+                files = precomputed_ds.get_fragments()[0].data_files()
+                if len(files) != 1:
+                    raise ValueError(
+                        "precomputed_partiton_dataset must have only one files"
+                    )
+                kwargs["precomputed_partitions_file"] = precomputed_partiton_dataset
```

Comment:
where is the rust code to process this kwargs

Reply:
https://github.com/lancedb/lance/blob/394360a17fc786e66bfec680cca6d03c4321c3b2/rust/lance/src/index/vector/ivf.rs#L1170-L1184

ww used this arg for GPU training already, this is just allowing an override